R/1.0_module_data_summary.R

```diff
@@ -0,0 +1,263 @@
+#' Data summary
+#' @description
+#' Module and its utils to display the number of rows and subjects in the filtered and unfiltered data.
+#'
+#' @details Handling different data classes:
+#' `get_object_filter_overview()` is a pseudo S3 method which has variants for:
+#' - `array` (`data.frame`, `DataFrame`, `array`, `Matrix` and `SummarizedExperiment`): Method variant
+#' can be applied to any two-dimensional objects on which [ncol()] can be used.
+#' - `MultiAssayExperiment`: for which summary contains counts for `colData` and all `experiments`.
+#'
+#' @param id (`character(1)`)
+#'  `shiny` module instance id.
+#' @param teal_data (`reactive` returning `teal_data`)
+#'
+#' @name module_data_summary
+#' @rdname module_data_summary
+#' @return `NULL`.
+NULL
+
+#' @rdname module_data_summary
+#' @keywords internal
+ui_data_summary <- function(id) {
+  ns <- NS(id)
+  tags$div(
+    id = id, # not used, can be used to customize CSS behavior
+    class = "well",
+    tags$div(
+      class = "row",
+      tags$div(
+        class = "col-sm-9",
+        tags$label("Active Filter Summary - teal", class = "text-primary mb-4")
+      ),
+      tags$div(
+        class = "col-sm-3",
+        actionLink(
+          ns("minimise_filter_overview"),
+          label = NULL,
+          icon = icon("angle-down", lib = "font-awesome"),
+          title = "Minimise panel",
+          class = "remove pull-right"
+        )
+      )
+    ),
+    tags$div(
+      id = ns("filters_overview_contents"),
+      tags$div(
+        class = "teal_active_summary_filter_panel",
+        tableOutput(ns("table"))
+      )
+    )
+  )
+}
+
+#' @rdname module_data_summary
+#' @keywords internal
+srv_data_summary <- function(id, teal_data) {
+  checkmate::check_class(teal_data, "reactive")
+  moduleServer(
+    id = id,
+    function(input, output, session) {
+      logger::log_trace("srv_data_summary initializing")
+
+      observeEvent(input$minimise_filter_overview, {
+        shinyjs::toggle("filters_overview_contents")
+        toggle_icon(session$ns("minimise_filter_overview"), c("fa-angle-right", "fa-angle-down"))
+        toggle_title(session$ns("minimise_filter_overview"), c("Restore panel", "Minimise Panel"))
```

Comment:
These two are private functions from teal.slice

Reply:
Including @m7pr to this:

Do we have any replacements in `teal` for this? Otherwise please copy `toggle_icon` and `toggle_title` to teal and place it in module_data_summary.R until we change the look of this panel/table

Reply:
Ah sorry guys. I though this is shiny in-built function. Will copy paste in a sec

Reply:
a8a386f339c4ad580e9e286c9c0b4c88b7e5f814